### PR TITLE
minor string handling optimization

### PR DESCRIPTION
**Pull request by moderakh**
* keeping activityId as a UUID in DSR to avoid the cost for translating from String to UUID and parsing back to UUID
* using StringUtils.split instead of split to avoid regex cost
* Cache consistency level enum with their string values to avoid parsing costs. as per my comparison (https://github.com/moderakh/azure-cosmosdb-benchmark) this is 300x faster than `EnumUtils.getEnum`

### CODE REVIEW

**Comment by kirankumarkolli** [comment]
>         final String value = request.getResourceId();

@David.Noble@microsoft.com /@mo what's the cost of this extraction?
Is it worth short specializing it?

---
Refers to: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestHeaders.java:994 in 7ea15c6. [](commit_id = 7ea15c66fedbc8d7c7ef5c7cdb037919c8ed979b, deletion_comment = False)

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/HttpTransportClient.java [hunk 1]
```diff
@@ -108,7 +108,7 @@ public void close() {
 
             ResourceOperation resourceOperation = new ResourceOperation(request.getOperationType(), request.getResourceType());
             // uuid correlation manager
-            String activityId = request.getActivityId();
+            String activityId = request.getActivityId().toString();
```
Comment (kirankumarkolli) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/HttpTransportClient.java]: Possible that in future there might be more string serilization consumers. Is it worth request having both byte[] or string part of Request?

Comment (moderakh) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/HttpTransportClient.java]: This is in my list to explore.
Libraries like avro have similar optimization:
https://github.com/rdblue/avro-java/blob/master/avro/src/main/java/org/apache/avro/util/Utf8.java

**Pull request merged**